### PR TITLE
Added Django and instructions for setup

### PR DESCRIPTION
**Pull request by ZhizhenQin**

### CODE REVIEW

File: ailab/benchmark/xdb.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+from __future__ import print_function
+from __future__ import unicode_literals
+import json
+
+# from .auth import context, interngraph_auth_params
+# from utils.custom_logger import getLogger
+# from utils.utils import requestsJson
+from db_utils import requestsJson
+
+DB_ENTRY = 'http://127.0.0.1:8000/benchmark/'
```
Comment (sf-wind) [inline ailab/benchmark/xdb.py]: User should be able to override it with a command line.

File: ailab/benchmark/xdb.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+NETWORK_TIMEOUT = 150
+
+
+class XDBDriver(object):
```
Comment (sf-wind) [inline ailab/benchmark/xdb.py]: Change the class name.

File: ailab/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+You can set up a server and and launch the benchmark run remotely.
+
+## Server Setup
+
+#### Config MySQL Database
+- Install MySQL on the server
+- Create a MySQL database
+- In the current directory (ailab), run
+- In `ailab/settings.py`, speficy `NAME`, `USER` and `PASSWORD` with the name
```
Comment (sf-wind) [inline ailab/README.md]: Need to install django?

File: ailab/benchmark/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                ('status', models.CharField(choices=[('AVA', 'AVAILABLE'), ('OCC', 'OCCUPIED'), ('OFF', 'OFFLINE'), ('DIS', 'DISABLED')], max_length=9, null=True)),
+                ('claimer', models.CharField(max_length=40, null=True)),
+                ('hash', models.CharField(max_length=32, null=True, unique=True)),
+                ('update_time', models.DateTimeField(null=True)),
+                ('heartbeat_time', models.DateTimeField(null=True)),
+                ('job_queue', models.CharField(default='pep_interactive', max_length=40)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Run',
```
Comment (sf-wind) [inline ailab/benchmark/migrations/0001_initial.py]: database name?

File: ailab/benchmark/xdb.py (hunk 2 above)
Comment (sf-wind) [reply ailab/benchmark/xdb.py]: Also need to change the file name.

Comment (sf-wind) [reply ailab/benchmark/xdb.py]: I don't think this file belongs here. It should be called in run_lab or run_remote.

File: ailab/benchmark/db_utils.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import json
+import os
+import requests
+import sys
+from time import sleep
+
+# from .custom_logger import getLogger
+
+
+def getBenchmarks(bfile, framework=None):
```
Comment (sf-wind) [inline ailab/benchmark/db_utils.py]: Is this method used here?

File: ailab/benchmark/db_utils.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            return result_json
+    except ValueError as e:
+        getLogger().error("Cannot decode json {}".format(e.output))
+
+    getLogger().error("Failed to retrieve json from {}".
+        format(url))
+    return {}
+
+
+def getPythonInterpreter():
```
Comment (sf-wind) [inline ailab/benchmark/db_utils.py]: Not used

File: ailab/benchmark/db_utils.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    return sys.executable
+
+
+def parse_kwarg(kwarg_str):
```
Comment (sf-wind) [inline ailab/benchmark/db_utils.py]: Not used.

File: ailab/benchmark/xdb.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+import json
+
+# from .auth import context, interngraph_auth_params
+# from utils.custom_logger import getLogger
+# from utils.utils import requestsJson
```
Comment (sf-wind) [inline ailab/benchmark/xdb.py]: remove comments

File: ailab/benchmark/db_utils.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
```
Comment (sf-wind) [inline ailab/benchmark/db_utils.py]: None of the method in this file is used.

File: ailab/benchmark/migrations/0001_initial.py [hunk 10]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.18 on 2019-02-19 20:21
```
Comment (sf-wind) [inline ailab/benchmark/migrations/0001_initial.py]: I wonder whether we should check in this file, or users should generate this file themselves. 

I feel, if this file is generated, it should not be in the same directory as the repo. Is it possible to specify a place to put the generated files?

Or, at least, ignore this directory in git.

File: ailab/README.md (hunk 3 above)
Comment (ZhizhenQin) [reply ailab/README.md]: Yes. Will add it

File: ailab/benchmark/migrations/0001_initial.py (hunk 10 above)
Comment (ZhizhenQin) [reply ailab/benchmark/migrations/0001_initial.py]: Done. Deleted the migration file and added them to .gitignore.

File: ailab/benchmark/db_utils.py (hunk 9 above)
Comment (ZhizhenQin) [reply ailab/benchmark/db_utils.py]: Removed the entire file from this commit.

File: ailab/benchmark/xdb.py (hunk 8 above)
Comment (ZhizhenQin) [reply ailab/benchmark/xdb.py]: Removed the entire file from this commit.

File: ailab/benchmark/db_utils.py (hunk 7 above)
Comment (ZhizhenQin) [reply ailab/benchmark/db_utils.py]: Removed the entire file from this commit.

File: ailab/benchmark/db_utils.py (hunk 6 above)
Comment (ZhizhenQin) [reply ailab/benchmark/db_utils.py]: Removed the entire file from this commit.

File: ailab/benchmark/db_utils.py (hunk 5 above)
Comment (ZhizhenQin) [reply ailab/benchmark/db_utils.py]: Removed the entire file from this commit.

File: ailab/benchmark/migrations/0001_initial.py (hunk 4 above)
Comment (ZhizhenQin) [reply ailab/benchmark/migrations/0001_initial.py]: Removed the file and added migrations files to gitignore.

File: ailab/benchmark/xdb.py (hunk 2 above)
Comment (ZhizhenQin) [reply ailab/benchmark/xdb.py]: Removed the entire file from this commit.

File: ailab/benchmark/xdb.py (hunk 1 above)
Comment (ZhizhenQin) [reply ailab/benchmark/xdb.py]: Removed the entire file from this commit. This should be handled later in run_lab and/or run_remote

File: ailab/ailab/settings.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'eo+li9#pw2im0xeq3@bv0h0ir^o36w&do%&3nm!!+!bxz5q9my'
```
Comment (sf-wind) [inline ailab/ailab/settings.py]: I don't think you can expose this. It is supposed to be a secret.

File: ailab/ailab/settings.py [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (sf-wind) [inline ailab/ailab/settings.py]: check the comment

File: ailab/README.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+## Server Setup
+
+#### Install DJango
+```
+pip install Django
+```
+
+#### Config MySQL Database
+- Install MySQL on the server
```
Comment (sf-wind) [inline ailab/README.md]: can you provide examples on how to install mysql on the server, and how to create the database?

File: ailab/README.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+- Create a MySQL database
+- In the current directory (ailab), run
```
Comment (sf-wind) [inline ailab/README.md]: run what?

File: ailab/ailab/settings.py [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+"""
+
+import os
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
```
Comment (sf-wind) [inline ailab/ailab/settings.py]: please check this doc.

File: ailab/ailab/settings.py (hunk 11 above)
Comment (ZhizhenQin) [reply ailab/ailab/settings.py]: Will implement a mechanism to generate a random secret key the first time the project is started.

File: ailab/README.md (hunk 14 above)
Comment (ZhizhenQin) [reply ailab/README.md]: Should swap this line with line 16. Will fix it in the next commit.

**Pull request closed without merging**